docs/credentials.md

```diff
@@ -135,11 +136,64 @@ CREDENTIAL_VAULT_ADDRESS=http://127.0.0.1:8200
 CREDENTIAL_VAULT_SECRET_BASE_PATH=secret
 ```
 
-## Configure both File Storage and Vault Storage
+## Configure Aws Secret Manager
 
-It is possible and sometimes useful to setup and use both `.credentials` file and vault for secret storage at the same time.
-In this case, the MFTF tests are able to read secret data at runtime from both storage options, but the local `.credentials` file will take precedence.
+Aws Secrets Manager offers secret management that supports:
+- Secret rotation with built-in integration for Amazon RDS, Amazon Redshift, and Amazon DocumentDB
+- Fine-grained policies and permissions
+- Audit secret rotation centrally for resources in the AWS Cloud, third-party services, and on-premises
 
+### Prerequisites
+- AWS account
+- AWS Secret Manger is created and configured
+- IAM User or Role is created
+
+### Store secrets in Aws Secret Manager
+
+#### Secrets format
+`Secret Name`, `Secret Key`, `Secret Value` are three key pieces of information to construct an Aws Secret. 
+`Secret Key` and `Secret Value` can be any content you want to secure, `Secret Name` must follow the format:
+
+```conf
+mftf/<VENDOR>/<SECRET_KEY>
+```
+
+```conf
+# Secret name for carriers_usps_userid
+mftf/magento/carriers_usps_userid
+
+# Secret key for carriers_usps_userid
+carriers_usps_userid
+
+# Secret name for carriers_usps_password
+mftf/magento/carriers_usps_password
+
+# Secret key for carriers_usps_password
+carriers_usps_password
+```
+
+### Setup MFTF to use Aws Secret Manager
+
+To use Aws Secret Manager, the Aws region to connect to is required. You can set it through environment variable [`CREDENTIAL_AWS_SECRET_MANAGER_REGION`][] in `.env`.
+
+MFTF uses the recommended [Default Credential Provider Chain][credential chain] to establish connection to Aws Secret Manager service. 
+You can setup credentials according to [Default Credential Provider Chain][credential chain] and there is no MFTF specific setup required. 
+Optionally, however, you can explicitly set Aws profile through environment variable [`CREDENTIAL_AWS_SECRET_MANAGER_PROFILE`][] in `.env`.
+
+```conf
+# Sample Aws Secret Manager configuration
+CREDENTIAL_AWS_SECRET_MANAGER_REGION=us-east-1
+CREDENTIAL_AWS_SECRET_MANAGER_PROFILE=default
+```
+
+## Configure multiple credential storage
+
+It is possible and sometimes useful to setup and use multiple credential storage at the same time.
+In this case, the MFTF tests are able to read secret data at runtime from all storage options, in this case MFTF use the following precedence:
+
+```
+.credentials File > HashiCorp Vault > Aws Secret Manager
```

Comment:
I had a question about why `Vault` precedes over `AWS secrets manager`? Couldn't find anything on this in the AC. If the build is configured to use vault, should it even read from  `AWS secrets manager` if it doesn't find the secret?

Reply:
As for `Vault` and `AwsSM`, there is no hard rule which one should be preferred. I thought `AwsSM` is the last one added and should be the last one in the list. This order only matters when all storages are configured. If only `AwsSM` is configured vault will not be used. Let me know if you have any concern on the order.

Reply:
Since this build only talks about local use, this seems to be be right order as AwsSM is newly added. Just confirming. CI/CD use story may change up things here + documentation.